assets/source/setup-guide/app/views/WizardApp.js

```diff
@@ -45,14 +45,26 @@ const WizardApp = () => {
 		wcSettings.pinterest_for_woocommerce.isBusinessConnected
 	);
 
+	const appSettings = useSettingsSelect();
+	const isDomainVerified = useSettingsSelect( 'isDomainVerified' );
+	const isTrackingConfigured = useSettingsSelect( 'isTrackingConfigured' );
+
+	//Subscribe history listening
+	useEffect( () => {
+		const unlisten = getHistory().listen( () => {
+			setCurrentStep( getQuery().step );
+		} );
+
+		//cleanup function to avoid memory leaks
+		return unlisten;
+	}, [] );
```

Comment:
If I understand correctly we could simplify the code doing the following steps:

1. We could remove  `const [ currentStep, setCurrentStep ] = useState();`  and  
 ```
useEffect( () => {
		const unlisten = getHistory().listen( () => {
			setCurrentStep( getQuery().step );
		} );
		//cleanup function to avoid memory leaks
		return unlisten;
	}, [] ); 
```


2. Replacing:
``` 
   if ( ! currentStep ) { 
 	setCurrentStep( getCurrentStep() ); 
 	return <Spinner />; 
 }
``` 
with `const currentStep = getCurrentStep();`

3. I guess that the function `updateQueryString( { step: key } ); `  will trigger a new rendering and `const currentStep ` will be assigned with the new step.

Let me know your thoughts. Thank you!

Reply:
> 3\. I guess that the function `updateQueryString( { step: key } ); `  will trigger a new rendering and `const currentStep ` will be assigned with the new step.

Yes, the client-side routing watches the URL changes and passes down parsed URL query to page components as one of props. So it will trigger re-render after updating to URL query.

`<WizardApp>` happens to be [one of the top-level page components](https://github.com/woocommerce/pinterest-for-woocommerce/blob/801bf25bb200d35f42c44583e1a4ff5a559bb318/assets/source/setup-guide/index.js#L65-L75), and the following code could inspect the `query` prop passing in from the routing handler.

```js
const WizardApp = ( { query } ) => {
	console.log( query );
	// ...omitted.
```

In other words, replacing `getQuery()` with `query` prop could further simplify this component. 😎

Reply:
Thanks, @eason9487 for the clarification! See the changes here: [82bae3d](https://github.com/woocommerce/pinterest-for-woocommerce/pull/318/commits/82bae3d774a2d1c43fe578d6e9c9e989d66b441b)